File: socketrunner_unity/Assets/Scripts/Commands/Tree1command.cs
using System;
using UnityEngine;

public class Tree1Command : ICommand
{
	private readonly bool _flag;
	private Tree1 _tree1;

	public Tree1Command(bool flag)
	{
		_flag = flag;
	}

	public void Execute()
	{
		_tree1?.setActiveTree1(_flag);
	}
}

File: server/src/commands/Tree1Command.js
const { Command } = require('./Command');

class Tree1Command extends Command {
    constructor(flag) {
        super();
        this._flag = true;
    }

    get type() {
        return 'TREE1_COMMAND';
    }

    switchFlag(){
        this._flag = !this._flag;
        return this._flag;
    }

    execute(ws) {

        if (!ws || ws.readyState !== 1) return false;
        console.log('reached execute tree1 js');
        ws.send(JSON.stringify({ type: this.type, payload: { flag: this.switchFlag() } }));


        return true;
    }
}

module.exports = { Tree1Command };
